src/test/java/test/dataprovider/DataProviderTest.java

```diff
@@ -95,8 +95,7 @@ public void allMethodsShouldBeInvoked() {
   public void failedDataProviderShouldCauseSkip() {
     InvokedMethodNameListener listener = run(DependentSample.class);
 
-    assertThat(listener.getSucceedMethodNames()).containsExactly("method1(ok)");
-    assertThat(listener.getSkippedMethodNames()).containsExactly("method2");
+    assertThat(listener.getSucceedMethodNames()).containsExactly("method1(ok)", "method2");
```

Comment:
Does this change expected? The sample has no retry analyzer and the `method2` should not be executed if I understand the sample well

Reply:
@juherr - Thanks. This was a bug in the code. I didn't realize that there was no retry involved, but it was a data provider. I have fixed this by differentiating between methods which get executed repeatedly (via a data provider or invocation count) and methods get retried.